### ISSUE TITLE
Faillure on nightly

### ISSUE THREAD

**Issue by lrnv** [julia-vscode/TestItemRunner.jl#64]
Hey, 

I have a faillure on nightly there https://github.com/lrnv/Copulas.jl/actions/runs/7903779228/job/21572481792 that seems to be a bit recurent. It says: 

```julia
ERROR: LoadError: MethodError: Cannot `convert` an object of type 
  TestItemRunner.Tokenize.Lexers.Lexer{Base.GenericIOBuffer{Memory{UInt8}},TestItemRunner.Tokenize.Tokens.RawToken} to an object of type 
  TestItemRunner.Tokenize.Lexers.Lexer{Base.GenericIOBuffer{Vector{UInt8}},TestItemRunner.Tokenize.Tokens.RawToken}
The function `convert` exists, but no method is defined for this combination of argument types.

Closest candidates are:
  (::Type{TestItemRunner.Tokenize.Lexers.Lexer{IO_t, T}} where {IO_t<:IO, T<:TestItemRunner.Tokenize.Tokens.AbstractToken})(::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any)
   @ TestItemRunner ~/.julia/packages/TestItemRunner/XLYqa/packages/Tokenize/src/lexer.jl:28
  convert(::Type{T}, ::T) where T
   @ Base Base.jl:126
```

Looks related to `Vector{}` vs `Memory{}` that was recently introduced ?

**Comment by tkemmer** [comment]
This issue also affects Julia 1.11-alpha1+ and has been fixed last week in the underlying CSTParser package (https://github.com/julia-vscode/CSTParser.jl/pull/385). Would it be possible to get a version bump for TestItemRunner.jl with the fixed CSTParser included?

**Comment by davidanthoff** [comment]
I just tagged a build that hopefully has this fixed. Can you try? And if it didn't solve it, reopen this issue? Thanks!

**Comment by lrnv** [comment]
Cannot test anymore since I decided to remove nightly testing from my CI setup after this one, makes no sense for me to catch these types of bugs since they are usually solved before releases of julia. 

Thanks a lot for the fix !

**Comment by tkemmer** [comment]
> I just tagged a build that hopefully has this fixed. Can you try? And if it didn't solve it, reopen this issue? Thanks!

This seems to have solved the issue, for both Julia 1.11 and nightly. Thanks a lot!
